tests/inject/dynamic/shadow-dom.tests.ts

```diff
@@ -88,4 +88,53 @@ describe('Handle Shadow-DOM', () => {
         expect(getComputedStyle(shadowRoot.querySelector('p')).color).toBe('rgb(255, 26, 26)');
     });
 
+    it('should handle defined elements', async () => {
+        container.innerHTML = multiline(
+            '<svg-container>',
+            '   <svg viewBox="0 0 100 30">',
+            '       <style class="testcase-style">#svg-text { color: red; }</style>',
+            '       <text id="svg-text" y="20">I am SVG</text>',
+            '   </svg>',
+            '</svg-container>',
+        );
+        class SVGContainer extends HTMLElement {
+            constructor() {
+                super();
+                const shadowRoot = this.attachShadow({mode: 'open'});
+                shadowRoot.append(...this.childNodes as any);
```

Comment:
Well, that was a bad example :) No need to copy anything, it's a logic for another purpose highlighted in that stack overflow answer. So, you can append some style to the shadow root inside the constructor.